packages/streamdown/lib/message-actions.tsx

```diff
@@ -0,0 +1,210 @@
+      )}
+    </button>
+  );
+};
+
+export const MessageExportDropdown = ({ content, className }: MessageActionsProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [isExporting, setIsExporting] = useState(false);
+
+  const handleMarkdownDownload = () => {
+    const timestamp = new Date().toISOString().slice(0, 16).replace(':', '-');
+    const filename = `message-${timestamp}.md`;
+    save(filename, content, "text/markdown");
+    setIsOpen(false);
+  };
+
+  const handlePdfExport = async () => {
+    setIsExporting(true);
+    try {
+      // Import jsPDF and html2canvas dynamically to avoid bundle bloat
+      const [{ default: jsPDF }, { default: html2canvas }] = await Promise.all([
+        import("jspdf") as Promise<{ default: any }>,
+        import("html2canvas") as Promise<{ default: any }>
+      ]);
+
+      // Create a temporary container with the markdown rendered as HTML
+      const tempDiv = document.createElement("div");
+      tempDiv.style.cssText = `
+        position: absolute;
+        top: -10000px;
+        left: -10000px;
+        width: 800px;
+        padding: 40px;
+        background: white;
+        color: black;
+        font-family: system-ui, -apple-system, sans-serif;
+        line-height: 1.6;
+      `;
+
+      // Configure marked for better PDF rendering
+      marked.setOptions({
+        breaks: true,
+        gfm: true,
+      });
+
+      // Convert markdown to HTML using marked
+      const htmlContent = marked.parse(content);
+
+      // Add styles to the HTML content for better PDF rendering
+      const styledHtmlContent = `
+        <style>
+          h1 { font-size: 24px; margin: 20px 0 10px 0; font-weight: bold; }
+          h2 { font-size: 20px; margin: 18px 0 8px 0; font-weight: bold; }
+          h3 { font-size: 16px; margin: 16px 0 6px 0; font-weight: bold; }
+          h4 { font-size: 14px; margin: 14px 0 6px 0; font-weight: bold; }
+          h5 { font-size: 12px; margin: 12px 0 6px 0; font-weight: bold; }
+          h6 { font-size: 10px; margin: 10px 0 6px 0; font-weight: bold; }
+          p { margin: 8px 0; line-height: 1.6; }
+          ul, ol { margin: 8px 0; padding-left: 20px; }
+          li { margin: 4px 0; }
+          code { background: #f5f5f5; padding: 2px 4px; border-radius: 3px; font-family: 'Monaco', 'Consolas', monospace; }
+          pre { background: #f5f5f5; padding: 10px; border-radius: 5px; margin: 10px 0; overflow-x: auto; }
+          pre code { background: none; padding: 0; }
+          blockquote { border-left: 4px solid #ddd; padding-left: 10px; margin: 10px 0; font-style: italic; }
+          table { border-collapse: collapse; width: 100%; margin: 10px 0; }
+          th, td { border: 1px solid #ddd; padding: 8px; text-align: left; }
+          th { background-color: #f5f5f5; font-weight: bold; }
+          strong { font-weight: bold; }
+          em { font-style: italic; }
+        </style>
+        ${htmlContent}
+      `;
```

Comment:
need to test out/match the latex and all forms of rendering done in the dom to be one to one in the exported content